### ISSUE TITLE
Upgrading from CF to SAM - Stack already exists error

### ISSUE THREAD

**Issue by what-name** [awslabs/serverless-application-model#1645]
**Description:**
I'm transitioning from a currently deployed pure CloudFormation stack to a SAM template. This will be the first time deploying to the stack with SAM. I configured my `samconfig.toml` to use the already deployed stack's name. When I do `sam deploy`, I get the following error:
```
Error: Failed to create changeset for the stack: my-backend-infra,
An error occurred (ValidationError) when calling the CreateChangeSet operation:
Stack [my-backend-infra] already exists and cannot be created again with the changeSet [samcli-deploy1593732248].
```

**Steps to reproduce the issue:**
1. Deploy pure CloudFormation template with the name `my-backend-infra`
2. Change (some) resources to SAM
3. `sam deploy --stack-name my-backend-infra`

**Observed result:**
SAM failed to update/deploy the `my-backend-infra` stack due to it already existing.

**Expected result:**
SAM updates the `my-backend-infra` stack in-place with the new resources.

In the [sam deploy docs](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-deploy.html) it says that the `--stack-name` flag will update the existing stack if it exists. I can not delete the current stack due to it containing other production resources.

**Comment by timoschilling** [comment]
Just a workaround: take a look on CloudFormation `"DeletionPolicy" : "Retain"` and CloudFormation imports. You can deplete the current stack but without deleting the production relevant resources and then import them into the new stack.

**Comment by what-name** [comment]
I found the root of the problem. It was very helpful to add the `--debug` flag to `sam deploy` as it prints out the issue much more descriptively. Turns out that the user who I used to deploy the template with, did not have some `cloudformation:` permissions that were required. I tested with a user who had admin rights and it went through. It was not a problem with the stack already existing, but a policy problem that it had `AccessDenied` to the currently deployed stack. I was able to migrate to SAM and update this stack in-place without any downtime.
